sys/ztimer/auto_init.c

```diff
@@ -52,78 +53,214 @@
 #define FREQ_1MHZ       1000000LU
 #define FREQ_250KHZ     250000LU
 #define FREQ_1KHZ       1000LU
+#define FREQ_1HZ        1LU
 
+/* Step 0: define available ztimer-periphery by activated modules */
+
+/* #if CONFIG_ZTIMER_USEC_TYPE_PERIPH_TIMER
+ * replaces #if MODULE_ZTIMER_PERIPH_TIMER
+ * the ztimer_periph_timer is always available
+ * having an (1) config defined in ztimer/config.h
+ */
+
+#if CONFIG_ZTIMER_USEC_TYPE_PERIPH_TIMER
+#  define ZTIMER_TIMER      _ztimer_periph_timer
+#  define ZTIMER_TIMER_CLK  _ztimer_periph_timer.super
+#endif
+
+#if MODULE_ZTIMER_PERIPH_RTT
+#  define ZTIMER_RTT      _ztimer_periph_timer_rtt
+#  define ZTIMER_RTT_CLK  _ztimer_periph_timer_rtt
+#endif
+
+#if MODULE_ZTIMER_PERIPH_RTC
+#  define ZTIMER_RTC      _ztimer_periph_timer_rtc
+#  define ZTIMER_RTC_CLK  _ztimer_periph_timer_rtc
+#endif
+
+/* Step 1: select which periphery to use for the higher level ZTIMER_*SEC
+ *         selected periphery is marked for initialisation (INIT_ZTIMER_<periph>
+ *         prepare defines for ztimer initialization
+ */
+
+/* ZTIMER_USEC always uses the basic timer
+ * basic timer is available on all boards */
 #if MODULE_ZTIMER_USEC
-#  if CONFIG_ZTIMER_USEC_TYPE_PERIPH_TIMER
-static ztimer_periph_timer_t _ztimer_periph_timer_usec = {
+#  ifndef INIT_ZTIMER_TIMER
+#    define INIT_ZTIMER_TIMER 1
+#  endif
+#endif
+
+/* ZTIMER_MSEC prefers ZTIMER_RTT (ztrimer_periph_timer_rtt)
+ * if it is available and runs at a frequency > 1kHz
+ * if not it falls back to use the basic timer */
+#if MODULE_ZTIMER_MSEC
+#  if defined(ZTIMER_RTT) && RTT_FREQUENCY >= FREQ_1KHZ
+#    define ZTIMER_MSEC_RTT 1
+#    ifndef INIT_ZTIMER_RTT
+#      define INIT_ZTIMER_RTT 1
+#    endif
+#    if RTT_FREQUENCY != FREQ_1KHZ
+#      define ZTIMER_MSEC_CONVERT_LOWER_FREQ RTT_FREQUENCY
+#    endif
+#  else
+#    define ZTIMER_MSEC_TIMER 1
```

Comment:
Could this look simpler if instead of having `ZTIMER_MSEC_%` you had only one `ZTIMER_MSEC_BACKEND` that is defined to the correct backend, so:

```suggestion
#    define ZTIMER_MSEC_BACKEND     ZTIMER_TIMER_CLK
```

and then below you could have:

```
ztimer_clock_t *const ZTIMER_MSEC_BASE = &ZTIMER_MSEC_BACKEND ;
```

I didn't try to adapt all the code, so maybe its not actually better, but if we can reduce the amount of macros and ifdefs it would be nice.